src/Slider.test.tsx

```diff
@@ -316,15 +309,62 @@ describe('UpsellSlider', () => {
     describe('initialSlideIndex', () => {
         it('Opens the slider with the initialSlide', async () => {
             render(<Slider initialSlideIndex={2}>
-                <span key={ 1 } data-testid="child-1"/>
-                <span key={ 2 } data-testid="child-2"/>
-                <span key={ 3 } data-testid="child-3"/>
-                <span key={ 4 } data-testid="child-4"/>
+                <span key={1} data-testid="child-1"/>
+                <span key={2} data-testid="child-2"/>
+                <span key={3} data-testid="child-3"/>
+                <span key={4} data-testid="child-4"/>
             </Slider>);
 
             await waitFor(() => {
                 expect(scrollToSpy).toHaveBeenCalled();
             });
         });
     });
+
+    describe('onSlide', function () {
+        it('calls the onSlide callback when an intersection occurs', () => {
+            const onSlideSpy = jest.fn();
+            render(<Slider onSlide={onSlideSpy}>
+                <span key={1}/>
+                <span key={2}/>
+                <span key={3}/>
+                <span key={4}/>
+            </Slider>);
+
+            const intersectionObserverInstance = getIntersectionObserverInstance();
+            const [intersectionCallback] = intersectionObserverInstance;
+            intersectionCallback([], mockIntersectionObserver.mock.instances[0]);
+
+            expect(onSlideSpy).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe('visible slide indexes', () => {
+        it('retrieves the first and last fully visible slide indices', () => {
+            const ref = React.createRef<API>();
+            render(<Slider ref={ref}>
+                <span key={1}/>
+                <span key={2}/>
+                <span key={3}/>
+                <span key={4}/>
+                <span key={5}/>
+            </Slider>);
+
+            const intersectionObserverInstance = getIntersectionObserverInstance();
+            const [intersectionCallback] = intersectionObserverInstance;
+
+            const slides = screen.getAllByRole('listitem');
+
+            intersectionCallback([
+                { intersectionRatio: 1, target: slides[0] } as unknown as IntersectionObserverEntry,
+                { intersectionRatio: 1, target: slides[1] } as unknown as IntersectionObserverEntry,
+                { intersectionRatio: 1, target: slides[2] } as unknown as IntersectionObserverEntry,
+                { intersectionRatio: .5, target: slides[3] } as unknown as IntersectionObserverEntry,
+                { intersectionRatio: 0, target: slides[4] } as unknown as IntersectionObserverEntry,
+            ], mockIntersectionObserver.mock.instances[0]);
+
+            expect(ref.current!.getFirstFullyVisibleSlideIndex()).toBe(0);
```

Comment:
You only have one testcase on this function using default state. There is no test on it being updated on slide or button movement.

Reply:
That's because they aren't directly related to each other. I thought about it but I think it should be an e2e test, not a unit test.